CoffeeShop/src/main/java/client/Demo.java

```
@@ -49,18 +48,20 @@ public static void main(String[] args) {
         // Create a new demo object
         Demo demo = new Demo();
 
-        itemReader = new ItemFileReader("menu.txt");
         try {
+            itemReader = new ItemFileReader("menu.txt");
             itemReader.readFile();
         } catch (FileNotFoundException e) {
-            throw new RuntimeException(e);
+            System.err.println("Error: menu.txt file not found.");
+            return;
```

I think we need some retry logic here? If we are no longer throwing an exception this will fail silently if it cannot find the file. I think if we implement retry logic or just keep it throwing a runtime exception as we can't really do much with the program if it cannot find menu.txt.

I think we should also display it in a pop up so the user is notified through the gui as well just so the user is made aware but don't necessarily worry about that here

This will need to use Faiz's logger as well but that is a future thing, don't worry about it for now

---

No I think this should be left as a runtime error. If the file cannot be read then the program should just fail. If there is no menu then we cannot have any items. This should be left as it was previously